Tests/Services/TasksServiceTests.cs

```diff
@@ -415,4 +408,412 @@ await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+  }
+
+  [Test]
+  public async Task ListTasksDetailedAsyncReturnsCorrectTaskDetailedPage()
+  {
+    var taskResponse = new ListTasksDetailedResponse
+                       {
+                         Tasks =
+                         {
+                           new TaskDetailed
+                           {
+                             Id        = "taskId1",
+                             Status    = ArmoniK.Api.gRPC.V1.TaskStatus.Completed,
+                             CreatedAt = Timestamp.FromDateTime(DateTime.UtcNow),
+                             StartedAt = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-5)),
+                             EndedAt   = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-1)),
+                             ExpectedOutputIds =
+                             {
+                               "outputId1",
+                             },
+                             DataDependencies =
+                             {
+                               "dependencyId1",
+                             },
+                             SessionId = "sessionId1",
+                           },
+                           new TaskDetailed
+                           {
+                             Id        = "taskId2",
+                             Status    = ArmoniK.Api.gRPC.V1.TaskStatus.Cancelling,
+                             CreatedAt = Timestamp.FromDateTime(DateTime.UtcNow),
+                             StartedAt = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-10)),
+                             EndedAt   = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-5)),
+                             ExpectedOutputIds =
+                             {
+                               "outputId2",
+                             },
+                             DataDependencies =
+                             {
+                               "dependencyId2",
+                             },
+                             SessionId = "sessionId1",
+                           },
+                         },
+                         Total = 1,
+                       };
+    var mockInvoker = new Mock<CallInvoker>();
+    var callInvoker = mockInvoker.SetupAsyncUnaryCallInvokerMock<ListTasksRequest, ListTasksDetailedResponse>(taskResponse);
+
+    var taskService = MockHelper.GetTasksServiceMock(callInvoker);
+
+    var paginationOptions = new TaskPagination
+                            {
+                              Page          = 1,
+                              PageSize      = 10,
+                              Filter        = new Filters(),
+                              SortDirection = SortDirection.Asc,
+                            };
+
+
+    var result = await taskService.ListTasksDetailedAsync(new SessionInfo("sessionId1"),
+                                                          paginationOptions)
+                                  .FirstOrDefaultAsync();
+
+
+    ClassicAssert.IsNotNull(result);
```

Comment:
Please use the [Contraint model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html) of assertions.